buildingmotif-app/src/app/pointlabel-parser/pointlabel-parser.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { HttpErrorResponse, HttpHeaders } from '@angular/common/http';
+import { Model } from '../types'
+import { Observable, throwError } from 'rxjs';
+import { catchError, retry } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PointlabelParserService {
+
+  constructor(private http: HttpClient) { }
+
+  parse(pointlabels: string, parsers: string): Observable<Model | any> {
+    const headers = {'Content-Type': "application/json"}
+
+    return this.http.post(`http://localhost:5000/parsers`, {point_labels: pointlabels, parsers}, {headers, responseType: 'json'})
+      .pipe(
+        retry(3), // retry a failed request up to 3 times
+        catchError(this.handleError) // then handle the error
+      );
+  }
+
+  private handleError(error: HttpErrorResponse) {
```

Comment:
Seeing this error handling code in each service makes me wonder if we should have a single BuildingMOTIF API service, or maybe a BuildingMOTIF service super class that all of our sub services can extend to reduce code copying.